src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,165 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useMatch, useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person, SortType } from '../types/Person';
+import { getPeople } from '../api';
+
+function filterPeople(
+  people: Person[],
+  sex: string | null,
+  query: string | null,
+  centuries: string[],
+): Person[] {
+  let filtered = [...people];
+
+  if (sex) {
+    filtered = filtered.filter(person => person.sex === sex);
+  }
+
+  if (query) {
+    filtered = filtered.filter(person => {
+      const { name, motherName, fatherName } = person;
+      const chekingString = (name + motherName + fatherName).toLowerCase();
+
+      return chekingString.includes(query.toLowerCase());
+    });
+  }
+
+  if (centuries.length > 0) {
+    const getCentury = (person: Person) => {
+      return Math.ceil(person.born / 100).toString();
+    };
+
+    filtered = filtered.filter(person => (
+      centuries.includes(getCentury(person))
+    ));
+  }
+
+  return filtered;
+}
+
+function sortPeople(
+  people: Person[],
+  sort: string | null,
+  order: string | null,
+) {
+  const sorted = [...people];
+
+  if (sort) {
+    switch (sort) {
+      case SortType.Name:
+      case SortType.Sex:
+        sorted.sort((prevPerson, nextPerson) => (
+          prevPerson[sort].localeCompare(nextPerson[sort])
+        ));
+        break;
+
+      case SortType.Born:
+      case SortType.Died:
+        sorted.sort((prevPerson, nextPerson) => (
+          prevPerson[sort] - nextPerson[sort]
+        ));
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (order === 'desc') {
+    sorted.reverse();
+  }
+
+  return sorted;
+}
+
+export const PeoplePage: React.FC = () => {
+  const match = useMatch('people/:slug');
+  const selectedPersonSlug = match?.params.slug;
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isPeopleLoading, setIsPeopleLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isNoPeople, setIsNoPeople] = useState(false);
```

Comment:
this variable is not needed here. try to use `people`, `isPeopleLoading` and `isError` without creating new state variable